src/main/java/com/slack/kaldb/server/KaldbLocalSearcher.java

```diff
@@ -76,7 +85,11 @@ public static SearchQuery fromSearchRequest(KaldbSearch.SearchRequest searchRequ
     // Set hits
     ArrayList<String> protoHits = new ArrayList<>(searchResult.hits.size());
     for (T hit : searchResult.hits) {
-      protoHits.add(JsonUtil.writeAsString(hit));
+      try {
+        protoHits.add(JsonUtil.writeAsString(hit));
+      } catch (JsonProcessingException e) {
+        throw new RuntimeException(e);
```

Comment:
Should a single failed `searchResult` conversion be a runtime exception? Should we look at a null / optional approach here instead?

Reply:
that is a fair point! I think if we move to the 1 gPRC call per chunk we want to throw error?

But if we don't then what you are suggesting is the right way forward